hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieTimeline.java

```diff
@@ -281,6 +290,21 @@
    */
   Option<byte[]> getInstantDetails(HoodieInstant instant);
 
+  /**
+   * Get WriteOperationType for instant.
+   */
+  Option<WriteOperationType> getWriteOperationType(HoodieInstant instant);
```

Comment:
It looks weird to have an API like this at  HoodieTimeline level; it's done via org.apache.hudi.common.model.HoodieCommitMetadata#getOperationType and the responsibility shouldn't fall on HoodieTimeline. At this stage, can we keep this internal to `isDeletePartitionType()` instead of making an API?